### ISSUE TITLE
turning off llocal and slocal lambda fails with div by 0 error

### ISSUE THREAD

**Issue by tjparnell** [taoliu/MACS#150]
If I run macs2 without a control bam file, and turn off local lambda (setting both llocal and slocal to 0), will crash macs2 during peak call. It appears llocal is more important, as setting slocal to 0 is ok. This happens with both single and paired-end bam files. Shouldn't the genome background (which I thought is always calculated) prevent a zero from occurring in the lambda control track?

Traceback (most recent call last):
  File "/home/BioApps/macs2/macs2", line 4, in <module>
    __import__('pkg_resources').run_script('MACS2==2.1.1.20160309', 'macs2')
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 744, in run_script
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 1506, in run_script
  File "/home/BioApps/python2.7.8/lib/python2.7/site-packages/MACS2-2.1.1.20160309-py2.7-linux-x86_64.egg/EGG-INFO/scripts/macs2", line 617, in <module>
    
  File "/home/BioApps/python2.7.8/lib/python2.7/site-packages/MACS2-2.1.1.20160309-py2.7-linux-x86_64.egg/EGG-INFO/scripts/macs2", line 57, in main
    
  File "build/bdist.linux-x86_64/egg/MACS2/callpeak_cmd.py", line 264, in run
  File "MACS2/PeakDetect.pyx", line 110, in MACS2.PeakDetect.PeakDetect.call_peaks (MACS2/PeakDetect.c:1665)
  File "MACS2/PeakDetect.pyx", line 321, in MACS2.PeakDetect.PeakDetect.__call_peaks_wo_control (MACS2/PeakDetect.c:3819)
ZeroDivisionError: float division by zero

**Comment by taoliu** [comment]
@tjparnell If you want to turn off local lambda, just use '--nolambda'.

**Comment by tjparnell** [comment]
oh, of course.

**Comment by kwcurrin** [comment]
I'm having an issue that I think is related to this. There is a short 17bp region in the middle of chr7 in one of my control bdg files that has a value of 0. I am using a treatment bed file as input with no control bam. I did not turn off slocal or llocal and I did not set nolambda. How is there a 0 value in the control bdg? Shouldn't the genome background be substitute if there aren't any nearby reads in that region?
